spec/features/admin_spec.rb

```diff
@@ -0,0 +1,194 @@
+  let(:random_token_uuid1) { "5373d24b-bfd0-4327-9762-70f8016f120e" }
+  let(:random_token_uuid2) { "77c24c90-01c3-4ef5-91e0-0fcc202ddcb9" }
+  let!(:random_user) do
+    create(:user,
+           context: valid_siret,
+           created_at: mardi_24_aout,
+           updated_at: mardi_24_aout,
+           email: "random@email.com",
+           id: random_user_uuid,
+           note: "some observations"
+          )
+  end
+
+  let!(:random_token1) do
+    create(:jwt_api_entreprise,
+           user: random_user,
+           id: random_token_uuid1,
+           created_at: mardi_24_aout,
+           updated_at: mardi_24_aout,
+           exp: Time.local(2022,2,24,12,0),
+           subject: "Simplification de démarches",
+          )
+  end
+
+  let!(:random_token2) { create(:jwt_api_entreprise, user: random_user, id: random_token_uuid2) }
+  let!(:archived_token)     { create(:jwt_api_entreprise, archived: true) }
+  let!(:not_archived_token) { create(:jwt_api_entreprise, archived: false) }
+
+  let!(:blacklisted_token)      { create(:jwt_api_entreprise, blacklisted: true) }
+  let!(:not_blacklisted_token)  { create(:jwt_api_entreprise, blacklisted: false) }
+
+  let!(:confirmed_user)      { create(:user, oauth_api_gouv_id: 12) }
+  let!(:unconfirmed_user)    { create(:user, oauth_api_gouv_id: nil) }
+
+  def dom_id(record)
+    "#" << [record.class.to_s.underscore, record.id].join('_')
+  end
+
+  it_behaves_like 'admin_path', "/admin/users"
+  it_behaves_like 'admin_path', "/admin/users/ccfa7701-1ee1-47b5-9af1-4e59dfa453d1"
+
+  it_behaves_like 'admin_path', "/admin/tokens"
+  it_behaves_like 'admin_path', "/admin/tokens/5373d24b-bfd0-4327-9762-70f8016f120e"
+
+  describe 'displays users list with generic information' do
+    before do
+      login_as(admin)
+      visit(admin_users_path)
+    end
+
+    it 'displays users in a table with one row per user' do
+      expect(page.all(".user_summary").size).to eq(User.count)
+    end
+
+    it 'email' do
+      within(dom_id(random_user)) do
+        expect(page).to have_content(random_user.email)
+      end
+    end
+
+    it 'context' do
+      within(dom_id(random_user)) do
+        expect(page).to have_content(random_user.context)
+      end
+    end
+
+    it 'created_at date in a readable fashion' do
+      within(dom_id(random_user)) do
+        expect(page).to have_content(random_user.created_at.strftime("%d/%m/%Y"))
+      end
+    end
+
+    it 'confirmed status as Non for unconfirmed user' do
+      within(dom_id(unconfirmed_user)) do
+        expect(page).to have_content('Non')
+      end
+    end
+
+    it 'confirmed status as Oui for confirmed user' do
+      within(dom_id(confirmed_user)) do
+        expect(page).to have_content('Oui')
+      end
+    end
+
+    it 'links to user details' do
+      within(dom_id(random_user)) do
+        expect(page).to have_link(random_user.email, href: admin_user_path(random_user))
+      end
+    end
```

Comment:
Tu as un helper à disposition pour tester l'affichage d'un tableau et son contenu : https://github.com/teamcapybara/capybara/blob/master/lib/capybara/spec/session/has_table_spec.rb 

Avec l'option `:with_rows` tu économises tous ces tests unitaires et la spec est plus solide imo. 

J'ai aussi une préférence pour l'utilisation de `create_list` pour créer tes records de user en base, et ensuite faire matcher les specs sur les attributs des deux ou trois users créés plutôt que sur un seul comme ici.

Reply:
roger that

Reply:
bon j ai essaye et c etait bien en theorie mais moins pratique a l'usage.

Reply:
C'est quoi le soucis ? Je trouve le helper très pratique au contraire.